From f69c3f892be7d2429e81c52aa2acd527516afa1e Mon Sep 17 00:00:00 2001
From: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Date: Tue, 6 Dec 2022 18:46:46 +0100
Subject: [PATCH 168/529] ARM: configs: use USB onboard HUB as built-in in
 multi_v7 addons fragment

Enable onboard hub driver as built-in, as this usbcore driver may report
some errors upon attach while booting and probing as a module. This
condition occurs when the HUB supply gets enabled by another onboard
peripheral. Then the HUB gets enumerated before the module gets loaded.
Symptom is an error message printed before the driver gets probed:
[   13.419482] Failed to attach USB driver: -22
[   13.430465] usbcore: registered new device driver onboard-usb-hub
It seems there's no impact, still rather use it as built-in.

Signed-off-by: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Change-Id: I29f24ad0a4f4b17fb3e69fb050355d5fdfcbb3f2
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/279507
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 arch/arm/configs/fragment-02-multiv7_addons.config | 1 +
 1 file changed, 1 insertion(+)

--- a/arch/arm/configs/fragment-02-multiv7_addons.config
+++ b/arch/arm/configs/fragment-02-multiv7_addons.config
@@ -48,6 +48,7 @@ CONFIG_THERMAL=y
 CONFIG_TMPFS=y
 CONFIG_TYPEC_UCSI=m
 CONFIG_UCSI_STM32G0=m
+CONFIG_USB_ONBOARD_HUB=y
 CONFIG_USELIB=y
 CONFIG_VFP=y
 CONFIG_ZBOOT_ROM_BSS=0x0
